### PR TITLE
op-dispute-mon: Expand acceptance coverage

### PR DESCRIPTION
**Pull request by ajsutton**
Builds on ethereum-optimism/optimism#21862.

Expands the dispute monitor acceptance suite across healthy and incorrect SuperPermissioned games, invalid proposal forecasting, reference-node failures, resolved bond accounting, and honest-actor losses. Adds reusable devstack metrics snapshot assertions and the minimal DSL/runtime wiring needed to exercise those scenarios.

Tested all six acceptance scenarios three times on each supported matrix: op-node/op-geth, op-node/op-reth, and kona-node/op-reth. Also ran `go test ./op-dispute-mon/... ./op-devstack/... ./op-service/testlog -count=1` and `mise exec -- just lint-go`.

### CODE REVIEW

File: op-devstack/presets/minimal.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ func (m *Minimal) AnchorStateRegistry() *dsl.AnchorStateRegistry {
 
 func (m *Minimal) AdvanceTime(amount time.Duration) {
 	m.T.Require().NotNil(m.timeTravel, "attempting to advance time on incompatible system")
-	m.timeTravel.AdvanceTime(amount)
+	m.L1EL.AdvanceTime(m.timeTravel, amount)
```
Comment (Inphi) [inline op-devstack/presets/minimal.go:61]: Codex AI has this to say:

This changes `Minimal.AdvanceTime` from clock-only advancement to waiting for an L1 block at the target timestamp. In `op-acceptance-tests/tests/batcher/batcher_test.go`, `sys.L1CL.Stop()` is called at line 57, followed by `sys.AdvanceTime(...)` at line 76 and manual L1 sequencing at line 80. Through the embedded `Minimal`, line 76 now calls `L1EL.AdvanceTime`, which waits for the block before execution can reach the manual sequencing step. With automatic CL production stopped, that wait cannot complete. The test is currently skipped, so CI does not expose this.
You should preserve the existing clock-only behavior of `Minimal.AdvanceTime` and perform the L1 block wait explicitly in the new dispute-monitor scenarios that require it.

File: op-devstack/devtest/metrics/metrics.go [hunk 2]
```diff
@@ -123,12 +125,110 @@ func parseSnapshot(payload string) (*Snapshot, error) {
...
+	}
+}
+
+func GaugeAtLeast(name string, labels map[string]string, minimum float64) SnapshotCheck {
+	return func(snapshot *Snapshot) error {
+		observed, err := snapshot.Gauge(name, labels)
+		if err != nil {
+			return err
+		}
+		if observed < minimum {
```
Comment (Inphi) [inline op-devstack/devtest/metrics/metrics.go:199]: Codex AI highlights that this check accidentally treats a `NaN` metric as meeting the minimum, because comparisons with `NaN` always return false. Could we explicitly reject `NaN` values here and add a test covering both a `NaN` metric value and a `NaN` minimum?

File: op-acceptance-tests/tests/interop/dispute-mon/dispute_mon_test.go [hunk 3]
```diff
@@ -1,19 +1,166 @@
...
+		sys.L1EL,
+		sys.L2Chain.DisputeGameFactoryProxyAddr(),
+		presets.WithDisputeMonRollupNodes(sys.L2CL),
+	)
+	mon.VerifyState(
+		disputemon.GameCount(gameTypes.CannonKonaGameType, 1),
+		disputemon.FailedGames(0),
+
+		disputemon.CompletedBeforeMaxDuration(1),
+		disputemon.ResolvedClaimsInFirstHalf(1),
```
Comment (Inphi) [inline op-acceptance-tests/tests/interop/dispute-mon/dispute_mon_test.go]: Another Codex finding:

> This test advances past the first-half boundary, but still expects the claim to be reported in the `first_half` bucket. It currently passes because the devstack and dispute monitor use different clocks.

Could we assert the total number of resolved claims here instead, regardless of the time bucket?

File: op-devstack/devtest/metrics/metrics.go (hunk 2 above)
Comment (ajsutton) [reply op-devstack/devtest/metrics/metrics.go:199]: hmm, my codex said that too and I told it to fix it... Apparently it didn't actually push the fix. :(

Comment (ajsutton) [reply op-devstack/devtest/metrics/metrics.go:199]: Pushed now.

File: op-devstack/presets/minimal.go (hunk 1 above)
Comment (ajsutton) [reply op-devstack/presets/minimal.go:61]: hmm, I'm going to disagree with this - advancing the L1 clock while L1 isn't running doesn't make any sense.  And that test is doomed because it seems to think sys.AdvanceTime will actually advance the L2 time which sadly isn't actually true.

File: op-acceptance-tests/tests/interop/dispute-mon/dispute_mon_test.go (hunk 3 above)
Comment (ajsutton) [reply op-acceptance-tests/tests/interop/dispute-mon/dispute_mon_test.go]: Switched to asserting the resolved claims and kicked off an agent to make dispute-mon respect time travel so we can test the timing functions better. That will be a follow up PR though.

**Review by Inphi — APPROVED** [review]
Excellent coverage!
